inference/synapse/synapse.go

```diff
@@ -105,20 +108,31 @@ func (s *Synapse) Close() {
 	log.Info("Synapse Engine Closed")
 }
 
-func (s *Synapse) InferByInfoHash(modelInfoHash, inputInfoHash string) ([]byte, error) {
+func CVMVersion(config *params.ChainConfig, num *big.Int) int {
+	// TODO(ryt): For Istanbul and versions after Istanbul, return CVM_VERSION_TWO
+	version := kernel.CVM_VERSION_ONE
+	if config.IsIstanbul(num) {
+		version = kernel.CVM_VERSION_TWO
+	}
+	return version
+}
+
+func (s *Synapse) InferByInfoHash(
+	modelInfoHash, inputInfoHash string, cvmVersion int) ([]byte, error) {
 	if s.config.IsRemoteInfer {
 		return s.remoteInferByInfoHash(modelInfoHash, inputInfoHash)
```

Comment:
Also version here ?

Reply:
yes, cause cvm-runtime need a version notation a bit different from CortexThesus Version like Byzantium and Istanbul. 

Since the CortexThesus version is forward compatible, that is A Istanbul version could also be regarded as Byzantium version, cvm-runtime only needs a notation that is unique and non-compatible for each other. 

Besides, we need to pass the version code to lower c++ via the interface. The single optional solution is passing the INT type as a version instead of a specific meaning string type(Byzantium, Istanbul, etc). 

Anyway, an int version is required in synapse version control.

Reply:
> Also version here ?

yes, cause cvm-runtime need a version notation a bit different from CortexThesus Version like Byzantium and Istanbul.

Since the CortexThesus version is forward compatible, that is A Istanbul version could also be regarded as Byzantium version, cvm-runtime only needs a notation that is unique and non-compatible for each other.

Besides, we need to pass the version code to lower c++ via the interface. The single optional solution is passing the INT type as a version instead of a specific meaning string type(Byzantium, Istanbul, etc).

Anyway, an int version is required in synapse version control.

Reply:
I mean we also need a version when ```s.remoteInferByInfoHash(modelInfoHash, inputInfoHash)``` @BetaBeauty

Reply:
you are right. @declmal, remote inference should also add version control. you can start a remote inference server to test this function. makefile have a infer\_server target to compile an infer server, more details refer to source code.